tf2onnx/onnx_opset/signal.py

```diff
@@ -0,0 +1,229 @@
+        perm = shapei[:-2] + [shapei[-1], shapei[-2]]
+        trx = ctx.make_node(
+            "Transpose", inputs=[node.input[0]], attr=dict(perm=perm),
+            name=utils.make_name(node.name + 'tr'))
+
+        ctx.remove_node(node.name)
+        mult = ctx.make_node(
+            "MatMul", inputs=[onx_real_imag_part.name, trx.output[0]],
+            name=utils.make_name('CPLX_' + node.name + 'rfft'))
+
+        new_shape = [2] + list(shape)
+        shapei = list(np.arange(len(new_shape)))
+        perm = shapei[:-2] + [shapei[-1], shapei[-2]]
+        last_node = ctx.make_node(
+            "Transpose", inputs=[mult.output[0]], attr=dict(perm=perm),
+            name=utils.make_name('CPLX_' + node.name + 'rfft'),            
+            shapes=[new_shape], dtypes=[res_onnx_dtype])
+
+        ctx.replace_all_inputs(node.output[0], last_node.output[0])  # ops=ctx.get_nodes()
+
+
+@tf_op("ComplexAbs")
+class ComplexAbsOp:
+    # support more dtype
+    supported_dtypes = [
+        onnx_pb.TensorProto.FLOAT,
+        onnx_pb.TensorProto.FLOAT16,
+        onnx_pb.TensorProto.DOUBLE,
+        onnx_pb.TensorProto.COMPLEX64,
+        onnx_pb.TensorProto.COMPLEX128,
+    ]
+
+    @classmethod
+    def any_version(cls, opset, ctx, node, **kwargs):
+        """
+        Computes the modules of a complex.
+        If the matrix dtype is not complex64 or complex128,
+        it assumes the first dimension means real part (0)
+        and imaginary part (1, :, :...).
+        """
+        onnx_dtype = ctx.get_dtype(node.input[0])
+        shape = ctx.get_shape(node.input[0])
+        np_dtype = utils.map_onnx_to_numpy_type(onnx_dtype)
+        utils.make_sure(shape[0] == 2, "ComplexAbs expected the first dimension to be 2 but shape is %r", shape)
+
+        ind0 = ctx.make_const(name=utils.make_name('cst0'), np_val=np.array([0], dtype=np.int64))
+        ind1 = ctx.make_const(name=utils.make_name('cst1'), np_val=np.array([1], dtype=np.int64))
+        p2 = ctx.make_const(name=utils.make_name('p2'), np_val=np.array([2], dtype=np_dtype))
+
+        real_part = ctx.make_node(
+            'Gather', inputs=[node.input[0], ind0.name], attr=dict(axis=0),
+            name=utils.make_name('Real_' + node.name))
+        imag_part = ctx.make_node(
+            'Gather', inputs=[node.input[0], ind1.name], attr=dict(axis=0),
+            name=utils.make_name('Imag_' + node.name))
+
+        real_part2 = ctx.make_node(
+            'Pow', inputs=[real_part.output[0], p2.name],
+            name=utils.make_name(real_part.name + 'p2p'))
+
+        imag_part2 = ctx.make_node(
+            'Pow', inputs=[imag_part.output[0], p2.name],
+            name=utils.make_name(imag_part.name + 'p2p'))
+
+        ctx.remove_node(node.name)
+        add = ctx.make_node(
+            "Add", inputs=[real_part2.output[0], imag_part2.output[0]],
+            name=utils.make_name('ComplexAbs_' + node.name))
+
+        if opset == 1:
```

Comment:
The change of axes (input vs attribute) occurs in opset 13, not opset 11.  The converter does not support opset 13 yet so you can always treat axes as an attribute.  I think this is why the unit tests are failing for opsets 11 and 12.

Reply:
fixed